implementations/rust/ockam/ockam_api/tests/portals.rs

```diff
@@ -0,0 +1,628 @@
+
+            let _outlet_status = second_node
+                .node_manager
+                .create_outlet(
+                    &second_node.context,
+                    echo_server_handle.chosen_addr,
+                    Address::from_string("outlet"),
+                    "outlet_alias".to_string(),
+                    true,
+                    Some(Arc::new(AllowAll)),
+                    None,
+                )
+                .await?;
+
+            let second_node_listen_address = second_node.listen_address().await;
+
+            // create inlet in the first node pointing to the second one
+            let inlet_status = first_node
+                .node_manager
+                .create_inlet(
+                    &first_node.context,
+                    "127.0.0.1:0".to_string(),
+                    "inlet_alias".to_string(),
+                    route![],
+                    route![],
+                    second_node_listen_address
+                        .multi_addr()?
+                        .concat(&MultiAddr::from_str("/secure/api/service/outlet")?)?,
+                    None,
+                    None,
+                    None,
+                    true,
+                )
+                .await?;
+
+            // connect to inlet_status.bind_addr and send dummy payload
+            let mut socket = TcpStream::connect(inlet_status.bind_addr.clone())
+                .await
+                .unwrap();
+            socket.write_all(b"hello").await.unwrap();
+
+            let mut buf = [0u8; 5];
+            socket.read_exact(&mut buf).await.unwrap();
+            assert_eq!(&buf, b"hello");
+
+            second_node.context.stop().await?;
+
+            // now let's verify the inlet has been detected as down
+            loop {
+                let inlet_status = first_node
+                    .node_manager
+                    .show_inlet("inlet_alias")
+                    .await
+                    .unwrap();
+                if inlet_status.status == ConnectionStatus::Down {
+                    break;
+                }
+                tokio::time::sleep(Duration::from_millis(5000)).await;
+            }
+
+            // create third node using the same address as the second one
+            let third_node =
+                TestNode::create(runtime, Some(&second_node_listen_address.to_string())).await;
+
+            let _outlet_status = third_node
+                .node_manager
+                .create_outlet(
+                    &third_node.context,
+                    echo_server_handle.chosen_addr,
+                    Address::from_string("outlet"),
+                    "outlet_alias".to_string(),
+                    true,
+                    Some(Arc::new(AllowAll)),
+                    None,
+                )
+                .await?;
+
+            // now let's verify the inlet has been restored
+            loop {
```

Comment:
Maybe worth limiting the number of iterations? If that part of the test fails - it'll just hang

Reply:
There is a timeout in case the test fails.